core/src/main/java/org/bitcoinj/evolution/CreditFundingTransaction.java

```diff
@@ -0,0 +1,212 @@
+        this.creditBurnIdentityIdentifier = Sha256Hash.ZERO_HASH;
+        if (creditBurnKey instanceof DeterministicKey) {
+            this.usedDerivationPathIndex = ((DeterministicKey)creditBurnKey).getChildNumber().num();
+        } else this.usedDerivationPathIndex = -1;
+        ScriptBuilder builder = new ScriptBuilder().addChunk(new ScriptChunk(OP_RETURN, null)).data(creditBurnPublicKey.getPubKeyHash());
+        lockedOutput = new TransactionOutput(params, null, fundingAmount, builder.build().getProgram());
+        addOutput(lockedOutput);
+    }
+
+    /**
+     * Creates a credit funding transaction by reading payload.
+     * Length of a transaction is fixed.
+     */
+
+    public CreditFundingTransaction(NetworkParameters params, byte [] payload) {
+        super(params, payload, 0);
+    }
+
+    /**
+     * Deserialize and initialize some fields from the credit burn output
+     */
+    @Override
+    protected void parse() throws ProtocolException {
+        super.parse();
+        parseTransaction();
+    }
+
+    @Override
+    protected void unCache() {
+        super.unCache();
+        lockedOutpoint = null;
+        lockedOutpoint = null;
+        creditBurnIdentityIdentifier = Sha256Hash.ZERO_HASH;
+        fundingAmount = Coin.ZERO;
+        creditBurnPublicKeyId = KeyId.KEYID_ZERO;
+    }
+
+    /**
+     * Initializes lockedOutput, lockedOutpoint, fundingAmount and the hash160
+     * credit burn key
+     */
+    private void parseTransaction() {
+        getLockedOutput();
+        getLockedOutpoint();
+        fundingAmount = lockedOutput.getValue();
+        getCreditBurnPublicKeyId();
+        creditBurnIdentityIdentifier = getCreditBurnIdentityIdentifier();
+    }
+
+    /**
+     * Sets lockedOutput and returns output that has the OP_RETURN script
+     */
+    public TransactionOutput getLockedOutput() {
+        if(lockedOutput != null)
+            return lockedOutput;
+
+        for(TransactionOutput output : getOutputs()) {
+            Script script = output.getScriptPubKey();
+            if(ScriptPattern.isCreditBurn(script)) {
+                lockedOutput = output;
+                return output;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Sets lockedOutpoint and returns outpoint that has the OP_RETURN script
+     */
+    public TransactionOutPoint getLockedOutpoint() {
+        if(lockedOutpoint != null)
+            return lockedOutpoint;
+
+        for(int i = 0; i < getOutputs().size(); ++i) {
+            Script script = getOutput(i).getScriptPubKey();
+            if(ScriptPattern.isCreditBurn(script)) {
+                //TODO:  This doesn't look correct, but matches dashsync-iOS.  Why is the txid reversed?
+                //having a reversed txid will not allow it to be searched or matched.
+                lockedOutpoint = new TransactionOutPoint(params, i, Sha256Hash.wrap(getTxId().getReversedBytes()));
```

Comment:
This doesn't look right, reversing bytes of transaction hash.  But it is matching the iOS implementation.

Reply:
based on comments from our iOS team the L2 functions of the network (dash platform) may be using the reverse byte order that Satoshi came up with for the bitcoin L1 (P2P protocol).  It drives us nuts every time we have to add code that uses hash values, so much big endian and little endian give us nightmares.